File: frontend/src/types/product.ts
export interface Product {
  id: number
  name: string
  description: string | null
  price: number
  image_url: string | null
  category: string
  stock: number
  created_at: string
  updated_at: string | null
}

export interface ProductCreate {
  name: string
  description?: string
  price: number
  image_url?: string
  category: string
  stock?: number
}

export interface ProductUpdate {
  name?: string
  description?: string
  price?: number
  image_url?: string
  category?: string
  stock?: number
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class ProductBase(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    image_url: Optional[str] = None
    category: str
    stock: int = 0


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    image_url: Optional[str] = None
    category: Optional[str] = None
    stock: Optional[int] = None


class Product(ProductBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
